app/controllers/events_controller.rb

```diff
@@ -129,9 +148,23 @@ def admin
   end
 
   def archive
-    @events = Event.past
-                   .paginate(page: params[:page], per_page: 20)
-                   .order("non_billig_start_time DESC")
+    events = Event
+             .active
+             .published
+             .past
+             .order('non_billig_start_time DESC')
+
+    @event_types = events
+                   .map { |e| [t("events.#{e.event_type}"), e.event_type] }
+                   .uniq
+                   .sort
+
+    @event_areas = events
```

Comment:
It may be better to user Area.all instead of iterating over all events. There are a lot of events in production